File: simple-stack-example-rx/src/main/java/com/zhuinden/stack_rx_example/SecondKey.java
package com.zhuinden.stack_rx_example;

import com.google.auto.value.AutoValue;

/**
 * Created by Owner on 2017. 02. 11..
 */

@AutoValue
public abstract class SecondKey extends Key {
    @Override
    public int layout() {
        return R.layout.path_second;
    }

    public static SecondKey create() {
        return new AutoValue_SecondKey();
    }
}

File: simple-stack-example-kotlin/src/main/kotlin/com/zhuinden/simplestackexamplekotlin/common/SecondKey.kt
package com.zhuinden.simplestackexamplekotlin.common

import com.zhuinden.simplestack.navigator.ViewChangeHandler
import com.zhuinden.simplestack.navigator.changehandlers.NoOpViewChangeHandler
import com.zhuinden.simplestackexamplekotlin.R
import paperparcel.PaperParcel
import paperparcel.PaperParcelable

/**
 * Created by Owner on 2017. 01. 12..
 */
@PaperParcel
data class SecondKey protected constructor(protected val tag: String = SecondKey::javaClass.name) //
    : PaperParcelable, Key {
    override fun layout(): Int = R.layout.path_second

    override fun viewChangeHandler(): ViewChangeHandler = NoOpViewChangeHandler()

    companion object {
        @JvmField val CREATOR = PaperParcelSecondKey.CREATOR

        fun create(): SecondKey {
            return SecondKey()
        }
    }
}
